libraries/persistent_store/src/format.rs

```diff
@@ -15,68 +15,104 @@
 // TODO(ia0): Remove when the module is used.
 #![allow(dead_code)]
 
-use crate::bitfield::*;
-use crate::{Storage, StorageIndex, StoreError, StoreResult};
+#[macro_use]
+mod bitfield;
+
+use self::bitfield::*;
+use crate::{usize_to_nat, Nat, Storage, StorageIndex, StoreError, StoreResult};
 use alloc::vec::Vec;
 use core::cmp::min;
+use core::convert::TryFrom;
 
+/// Internal representation of a word in flash.
+///
+/// Currently, the store only supports storages where a word is 32 bits.
 type WORD = u32;
 
+/// Abstract representation of a word in flash.
+///
+/// This type is kept abstract to avoid possible confusion with `Nat` if they happen to have the
+/// same representation. This is because they have different semantics, `Nat` represents natural
+/// numbers while `Word` represents sequences of bits (and thus has no arithmetic).
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+pub struct Word(WORD);
```

Comment:
Why is there a `WORD` and a `Word`? You don't need that for i.e. `Nat`. What if you just had a

> pub type Word = u32;

?

Reply:
It's because I want to have `Word` (and `Position`) abstract to avoid confusing them by mistake with `u32` because many of the operations that are possible on `u32` should be forbidden on those types (for modules that only see the public interface).

I gave a name to the internal representation of `Word` (i.e. I did not write `pub struct Word(u32)`) because it's used a few times in this module and writing `WORD` documents better the intent than writing `u32`.

The reason I did not do it for `Nat` is because it's more convenient. Many of the operations on `u32` are also applicable to `Nat` so I don't need to reimplement them (like I did for `Position`). And the risk to not have it abstract is low, because there is no other "`u32`" in the crate with which `Nat` could be confused.

Reply:
Makes sense!